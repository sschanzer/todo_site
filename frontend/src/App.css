.checkHolder {
  display: flex;
  justify-content: center;
  align-items: top;
  position: relative;
}
.check {
  position: absolute;
  opacity: 0;
  z-index: 2;
}
.alterBtn {
  height: 3vh;
  width: 2vw;
  font-size: 2vh;
  text-align: center;
  display: flex !important;
  justify-content: center;
  align-items: center;
}

.buttonHolder {
  display: flex;
}

.buttonImg {
  height: 2vh;
}
