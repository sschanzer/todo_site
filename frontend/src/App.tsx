import React, { useState, ChangeEvent, useEffect } from "react";
import "bootswatch/dist/darkly/bootstrap.min.css";
import "./App.css";
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import { InputGroup } from "react-bootstrap";
import Header from "./components/Header";
import axios from "axios";
import { Tasks } from "./components/Tasks";
import { CSRFToken } from "./components/CSRFToken";

export interface myTasks {
  id: number;
  title: string;
  completed: boolean;
}

export interface responseTasks {
  completed: myTasks[];
  pending: myTasks[];
}

export async function getTasks(): Promise<responseTasks> {
  let response = await axios.get("all_tasks");
  return response.data;
}

function App() {
  const [selectedTask, setSelectedTask] = useState<number[]>([]);
  const [completedTasks, setCompletedTasks] = useState<myTasks[]>([]);
  const [pendingTasks, setPendingTasks] = useState<myTasks[]>([]);

  useEffect(() => {
    const getResponseFromGetTasks = async () => {
      let response = await getTasks();
      setCompletedTasks(response.completed);
      setPendingTasks(response.pending);
    };
    getResponseFromGetTasks();
  }, []);

  CSRFToken();

  return (
    <div className="App">
      <Header pendingTasks={pendingTasks} setPendingTasks={setPendingTasks} />
      <Row style={{ textAlign: "center" }}>
        <h3>Pending Tasks</h3>
        {pendingTasks.map((task) => (
          <Tasks
            task={task}
            selected={selectedTask}
            setSelected={setSelectedTask}
          />
        ))}
      </Row>

      <Row style={{ textAlign: "center" }}>
        <h3>Completed Tasks</h3>
        {completedTasks.map((task) => (
          <Tasks
            task={task}
            selected={selectedTask}
            setSelected={setSelectedTask}
          />
        ))}
      </Row>
    </div>
  );
}

export default App;
