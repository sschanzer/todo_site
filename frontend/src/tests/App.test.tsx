import { describe, expect, it, vi, test } from "vitest";
import TestRenderer from "react-test-renderer";
import { Mocked } from "vitest";
import axios from "axios";
import { getTasks } from "../App";
import App from "../App";

vi.mock("axios");

describe("App", () => {
  describe("getTasks()", () => {
    test("returns tasks from database", async () => {
      const mockAxios = axios as Mocked<typeof axios>;
      mockAxios.get.mockResolvedValue({ data: { completed: [], pending: [] } });
      const tasks = await getTasks();
      expect(tasks).toStrictEqual({ completed: [], pending: [] });
    });
    it("will take a snapshot of the header", () => {
      const testApp = TestRenderer.create(<App />);
      expect(testApp).toMatchSnapshot();
    });
  });
});
