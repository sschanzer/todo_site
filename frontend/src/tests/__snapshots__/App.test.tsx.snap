// Vitest Snapshot v1

exports[`App > Make sure nothing changes in my css or html > will take a snapshot of the header 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-2"
        >
          <button
            className="btn btn-primary btn-sm"
          >
            Delete
          </button>
        </div>
        <div
          className="col-2"
        >
          <button
            className="btn btn-primary btn-sm"
          >
            Status
          </button>
        </div>
        <div
          className="col-1"
        >
          <label
            className="form-check-label"
          >
            P
          </label>
        </div>
        <div
          className="col-2"
        >
          <div
            className="form-check form-switch"
          >
            <input
              className="form-check-input"
              id="flexSwitchCheckDefault"
              role="switch"
              type="checkbox"
            />
          </div>
        </div>
        <div
          className="col-1"
        >
          <label
            className="form-check-label"
          >
            C
          </label>
        </div>
        <div
          className="col-4"
        >
          <div
            className="input-group mb-3"
          >
            <input
              className="form-control"
              id="inputDefault"
              placeholder="Task"
              type="text"
            />
            <button
              className="btn btn-primary"
              id="button-addon2"
              type="button"
            >
              +
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
    style={
      {
        "textAlign": "center",
      }
    }
  >
    <h3>
      Pending Tasks
    </h3>
  </div>
  <div
    className="row"
    style={
      {
        "textAlign": "center",
      }
    }
  >
    <h3>
      Completed Tasks
    </h3>
  </div>
</div>
`;
